<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>473</width>
    <height>202</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>410</width>
    <height>140</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="mainGridLayout">
    <item row="4" column="0">
     <widget class="QPushButton" name="generateBtn">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QLineEdit" name="randomLine">
      <property name="styleSheet">
       <string notr="true">font: 14pt &quot;FreeSerif&quot;;</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QComboBox" name="algorithmCombo">
      <property name="toolTip">
       <string>Ordered by reccomended use.
Random.org will grab data from random.org using https</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="algorithmLabel">
      <property name="text">
       <string>Algorithm:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <layout class="QHBoxLayout" name="lengthLayout">
      <item>
       <widget class="QSpinBox" name="lengthSpin">
        <property name="toolTip">
         <string>8 characters or more is recommended </string>
        </property>
        <property name="value">
         <number>8</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="lengthLabel">
      <property name="text">
       <string>Length:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="charLabel">
      <property name="text">
       <string>Characters:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <layout class="QHBoxLayout" name="charCheckLayout">
      <item>
       <widget class="QCheckBox" name="lettersCheck">
        <property name="toolTip">
         <string>a-zA-Z</string>
        </property>
        <property name="text">
         <string>Letters</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="numbersCheck">
        <property name="toolTip">
         <string>0-9</string>
        </property>
        <property name="text">
         <string>Numbers</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="punctCheck">
        <property name="toolTip">
         <string>,.!_@$^&amp;*()_</string>
        </property>
        <property name="text">
         <string>Punctuation</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="customCheck">
        <property name="toolTip">
         <string>Create you own custom character set</string>
        </property>
        <property name="text">
         <string>Custom</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="1">
     <widget class="QLineEdit" name="customCharSetLine">
      <property name="enabled">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
